Sources/XiEditor/LineCache.swift

```diff
@@ -110,13 +140,30 @@ fileprivate class LineCacheState<T>: UnfairLock {
     /// Updates the state by applying a delta. The update format is detailed in the
     /// [xi-core docs](https://github.com/google/xi-editor/blob/master/doc/update.md).
     func applyUpdate(update: [String: AnyObject]) -> InvalSet {
+        // parse annotations
+        let annotationData = (update["annotations"] as! [[String: AnyObject]])
+
+        for annotationType in AnnotationType.all {
+            let annotationsOfType = annotationData.filter({$0["type"] as! String == annotationType.rawValue})
+
+            if annotationsOfType.isEmpty {
+                annotations[annotationType] = []
+            } else {
+                let annotationTypeData = (annotationsOfType.first!)["data"] as! [[String: AnyObject]]
+                annotations[annotationType] = annotationTypeData.map({(d: [String: AnyObject]) -> Annotation in
```

Comment:
There's a super clever way to simplify this one ;)
`annotations[annotationType] = annotationTypeData.map(Annotation.init)`